parts/dcosmasterresources.t

```diff
@@ -141,9 +141,8 @@
         "name": "nicLoopNode"
       },
       "dependsOn": [
-{{if .MasterProfile.IsCustomVNET}}
         "[variables('masterNSGID')]",
-{{else}}
+{{if not .MasterProfile.IsCustomVNET}}        
```

Comment:
ARM uses the depends on to block creation of the resource that the variable represents.  For example, if variables('vnetID'), has not been created yet, then it will wait before creating the NIC.  However, in the bring your own VNET case, we are not creating a VNET, so we don't need to depends on clause in this case.

Interestingly as I was testing this, I forgot to add this in the depends and I actually got the following error from ARM, because it tried to create the NIC before the VNET:

        "message": "Resource /subscriptions/MYSUBIDREMOVED/resourceGroups/AHWE0124F/providers/M
icrosoft.Network/virtualNetworks/DCOS-VNET-35433090 not found."